src/json.hpp

```diff
@@ -2818,22 +2820,21 @@ class lexer
     std::size_t chars_read = 0;
     /// the start position of the current token
     std::size_t start_pos = 0;
-
     /// buffer for variable-length tokens (numbers, strings)
-    std::vector<char> yytext = std::vector<char>(1024, '\0');
+    std::vector<value_t> yytext = std::vector<value_t>(1024, '\0');
     /// current index in yytext
     std::size_t yylen = 0;
 
     /// a description of occurred lexer errors
-    const char* error_message = "";
+    const value_t* error_message = "";
```

Comment:
This can cause failures if the current string literals that are assigned to `error_message` are not compatible with `const value_t *`.